sstables/compaction_strategy.cc

```diff
@@ -295,8 +155,8 @@ class partitioned_sstable_set : public sstable_set_impl {
         }
     }
     // SSTables are stored separately to avoid interval map's fragmentation issue when level 0 falls behind.
-    bool store_as_unleveled(const shared_sstable& sst) const {
-        return _use_level_metadata && sst->get_sstable_level() == 0;
+    bool store_as_unleveled(const versioned_sstable& sst) const {
+        return _use_level_metadata && sst->sst->get_sstable_level() == 0;
```

Comment:
That might be problematic because the versioned_sstable is just lw_shared_ptr<versioned_sstable_data> where versioned_sstable_data is a struct with: shared_sstable sst; uint64_t version_added; std::optional<uint64_t> ersion_erased;
Maybe it isn't as bad as it sounds though, the "->sst->" only happens in 5 places and the similarly confusing "*sst->sst" in 2 places.